src/components/Identity/index.js

```diff
@@ -83,12 +80,12 @@ const createIdentity = ({ config, logger, cookieJar }) => {
       // Waiting for opt-in because we'll be reading the ECID from a cookie
       onBeforeEvent(event) {
         return optIn.whenOptedIn().then(() => {
-          const nowInHours = Math.round(
-            convertTimes(MILLISECOND, HOUR, new Date().getTime())
-          );
-          const timestamp = parseInt(cookieJar.get(ID_SYNC_TIMESTAMP) || 0, 36);
-
-          if (config.idSyncsEnabled && nowInHours > timestamp) {
+          if (
+            !alreadyQueriedForIdSyncs &&
+            config.idSyncsEnabled &&
+            idSyncs.hasExpired()
+          ) {
+            alreadyQueriedForIdSyncs = true;
```

Comment:
This logic means that you only query for IdSyncs once per page load.  Is this what we want for long running pages?  You could rename this variable to `idSyncsInFlight`. and then reset it when the query sync comes back.

Reply:
Jon, we sync once every 7 days.